app/uk/gov/hmrc/personaldetailsvalidation/endpoints/PersonalDetailsCollectionController.scala

```diff
@@ -27,11 +27,11 @@ class PersonalDetailsCollectionController @Inject()(page: PersonalDetailsPage,
                                                     personalDetailsSubmission: FuturedPersonalDetailsSubmission)
   extends FrontendController {
 
-  def showPage(implicit completionUrl: CompletionUrl): Action[AnyContent] = Action { implicit request =>
-    Ok(page.render)
+  def showPage(implicit completionUrl: CompletionUrl, alternativeVersion: Boolean = false): Action[AnyContent] = Action { implicit request =>
+    Ok(page.render(alternativeVersion))
   }
 
-  def submit(completionUrl: CompletionUrl): Action[AnyContent] = Action.async { implicit request =>
-    personalDetailsSubmission.bindValidateAndRedirect(completionUrl)
+  def submit(completionUrl: CompletionUrl, alternativeVersion: Boolean = false): Action[AnyContent] = Action.async { implicit request =>
+    personalDetailsSubmission.bindValidateAndRedirect(completionUrl, alternativeVersion)
```

Comment:
Do we need to pass query param here? The form body has sufficient information already with which we can find out if user submitted the postcode form or the nino form.